### PR TITLE
UCR report filter cleanup

### PR DESCRIPTION
**Pull request by czue**
no functional changes, just reacquainting myself with this code before diving into some work and felt the need to do a bit of housekeeping

can be read commit-by-commit. is mostly just moving stuff around.

@nickpell might be interested. cc @calellowitz

### CODE REVIEW

File: corehq/apps/userreports/filters/dynamic_choice_lists.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        query = query.filter(sql_column.contains(search_term))
+
+    offset = page * limit
+    try:
+        return [{'value': v[0], 'display': v[0]}
+                for v in query.distinct().order_by(sql_column).limit(limit).offset(offset)]
+    except ProgrammingError:
+        return []
+
+
```
Comment (dimagimon) [inline corehq/apps/userreports/filters/dynamic_choice_lists.py]: W391 blank line at end of file

File: corehq/apps/userreports/reports/filters/specs.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        ))
+
+
+class FilterSpec(JsonObject):
+    """
+    This is the spec for a report filter - a thing that should show up as a UI filter element
+    in a report (like a date picker or a select list).
+    """
+    type = StringProperty(required=True, choices=['date', 'numeric', 'choice_list', 'dynamic_choice_list'])
+    slug = StringProperty(required=True, validators=_validate_filter_slug)  # this shows up as the ID in the filter HTML.
```
Comment (dimagimon) [inline corehq/apps/userreports/reports/filters/specs.py]: E501 line too long (121 > 115 characters)

File: corehq/apps/userreports/reports/filters/specs.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    datatype = DataTypeProperty(default='string')
+
+    @property
+    def choices(self):
+        return []
+
+
+class NumericFilterSpec(FilterSpec):
+    type = TypeProperty('numeric')
+
```
Comment (dimagimon) [inline corehq/apps/userreports/reports/filters/specs.py]: W391 blank line at end of file

File: corehq/apps/reports_core/static/reports_core/js/choice-list-api.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+    module.getApiQueryParams = function (term, page) {
+        return {
+            q: term, // search term
+            page: page,
+            limit: pageSize
+        };
+    };
+    module.formatValueForSelect2 = function (val) {
+        return {'id': val.value, 'text': val.display}
```
Comment (dimagimon) [inline corehq/apps/reports_core/static/reports_core/js/choice-list-api.js]: Missing semicolon.

**Comment by czue** [comment]
:white_check_mark:

**Comment by czue** [comment]
@esoergel this is a bit of a prefactor for the location filter

**Comment by nickpell** [comment]
such prefactor very quality

**Pull request merged**